### PR TITLE
Make CI green

### PR DESCRIPTION
**Pull request by artonge**
- [x] Import and use needed linkify packages
- [x] Support npm run serve
- [x] Properly cast `$level` in `lib/AP.php`
- [x] Other psalm errors
- [x] Cypress tests

Fix https://github.com/nextcloud/social/issues/1557

### CODE REVIEW

**Comment by gbraad** [comment]
You might also wanna change `linkify` to `^4.1.0`

File: src/components/TimelinePost.vue [hunk 1]
```diff
@@ -78,11 +78,10 @@
 </template>
 
 <script>
+// eslint-disable-next-line no-unused-vars
 import * as linkify from 'linkifyjs'
-// eslint-disable-next-line
-import pluginMention from 'linkifyjs/plugins/mention'
-// eslint-disable-next-line
-import 'linkifyjs/string'
+import 'linkify-plugin-mention'
```
Comment (gbraad) [inline src/components/TimelinePost.vue:83]: I missed that. Now understand the error I got after adding the deps. 👍

**Comment by artonge** [comment]
> You might also wanna change `linkify` to `^4.1.0`

That's a minor release compared to our current version. So will let dependabot handle that one :)

File: package.json [hunk 2]
```diff
@@ -39,7 +41,10 @@
 		"@nextcloud/stylelint-config": "^2.3.0",
 		"@nextcloud/vue": "^7.3.0",
 		"@nextcloud/vue-richtext": "^2.0.4",
+		"cypress": "^12.3.0",
 		"he": "^1.2.0",
+		"linkify-plugin-mention": "^4.1.0",
+		"linkify-string": "^4.1.0",
```
[the remark this answers is not in the document]
Comment (gbraad) [reply package.json:47]: https://github.com/nextcloud/social/pull/1558#issuecomment-1371063108

Resolved

**Comment by artonge** [comment]
@ArtificialOwl can you take a look at the psalm errors ? fix them or hide them ?

**Comment by ArtificialOwl** [comment]
@artonge any idea for the last psalm entry ?

**Review by ArtificialOwl — APPROVED** [review]

**Comment by artonge** [comment]
Thanks @ArtificialOwl !

**Pull request merged**